src/targets/gpu/compile_ops.cpp

```diff
@@ -228,10 +228,12 @@ struct compile_plan
         return *results[i];
     }
 
-    void replace(module& m) const
+    instruction_ref
+    replace(module& m,
+            const std::unordered_map<instruction_ref, instruction_ref>& inputs_rep_map) const
     {
         const auto& cr = benchmark();
-        cr.replace.replace(m, cr.ins);
+        return cr.replace.replace(m, cr.ins, inputs_rep_map);
```

Comment:
Why are we changing this to take a map and return an instruction?

Reply:
It is because when compiling mlir dot with split-k config, it would split the input module into two. one with dot and another with pointwise. Split happens when compiling the original instruction. During splits "mod_with_inputs" is cached. This `mod.inputs` would refer to input instructions from original input program to "compile_ops" pass e.g. "precompile_ops". 

Compile happens in parallel. 

Replace happens after everything is compiled. I can't use older cached "precompile_ops" inputs to the dot and pointwise module during the replace. Older instructions are not removed untill Deadcodeelimination runs.  Therefore i need to maintain a map to use the new compiled instructions as the inputs to the "dot" and "pointwise" modules.

Reply:
But you have the inputs of the replaced compiled ops as you can get that with `ins->inputs()`. So you just need to map cached inputs from the split module to the new `ins->inputs()`, and this map can be captured by the lambda(so it doesnt need to be passed as a parameter). You dont even need to store a map, just the original `ins->inputs()` vector, and then you can create a map in the replace function using the two vectors.

Reply:
Good idea. Pushed changes. Thanks